pkg/solver/matcher/matcher.go

```diff
@@ -84,7 +76,7 @@ func GetMatchingDeals(
 			continue
 		}
 
-		// loop over resource offers
+		// Build list of matching resource offers
 		matchingResourceOffers := []data.ResourceOffer{}
 		for _, resourceOffer := range resourceOffers {
```

Comment:
Maybe we can have this loop be a private function that does all the magic and returns the populated `matchingResourceOffers` array to help shrink down the overall function size and help readability

Reply:
Let's hold off on this one for now. Refactoring into private functions is a bit noisy at the moment because we need to pass the `db`, `tracer`, `log`, etc to each function that uses them.

We can improve this situation by adding a `matcher` struct and implementing these functions as methods on the struct. That way the struct has all the things and we pass less around. Will open an issue to track this.

Reply:
Opened an issue for it: https://github.com/Lilypad-Tech/lilypad/issues/504